graphql/schema/response.go

```diff
@@ -134,15 +134,24 @@ func (r *Response) Output() interface{} {
 			Errors: []byte(`[{"message": "Internal error - no response to write."}]`),
 			Data:   []byte("null"),
 		}
-	}
-	return struct {
-		Errors     []*x.GqlError   `json:"errors,omitempty"`
-		Data       json.RawMessage `json:"data,omitempty"`
-		Extensions *Extensions     `json:"extensions,omitempty"`
-	}{
-		Errors:     r.Errors,
-		Data:       r.Data.Bytes(),
-		Extensions: r.Extensions,
+	} else if x.Config.GraphqlExtension {
+		return struct {
```

Comment:
You can simplify this.

You can have the struct with all 3 fields. 

And if flag is turned on, set the Extensions. Otherwise, it will be nil and will be omitted (because of the "omitempty" against the field) anyways in the JSON response.